lib/publiccloud/provider.pm

```diff
@@ -343,6 +345,10 @@ C<instance_type> defines the flavor of the instance. If not specified, it will l
 =cut
 
 sub create_instance {
+    my $max = get_var('PUBLIC_CLOUD_MAX_INSTANCES', 1);
+    if ($max > 0 && ++$instance_counter > $max) {
+        die "Method create_instance already called $instance_counter times!";
```

Comment:
```suggestion
        die "Maximum number of instances reached ($instance_counter times)";
```
Nitpick: Error messages are typically better understood if they are high-level.

Reply:
Fixed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20721/commits/6d6e3a4f154ba36f102eea47dc3982dbec048ae6. Thanks for the suggestion!